src/main/java/controller/ProductController.java

```diff
@@ -0,0 +1,124 @@
+package controller;
+
+import dto.mapper.DtoMapper;
+import dto.request.ProductRequestDto;
+import dto.response.ProductResponseDto;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import model.Product;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import service.ProductService;
+import util.SortUtil;
+
+@AllArgsConstructor
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final DtoMapper<ProductRequestDto, Product, ProductResponseDto> dtoMapper;
+    private final SortUtil sortUtil;
+
+    @GetMapping("/{id}")
+    @Operation(summary = "Get product by id")
+    public ProductResponseDto get(@PathVariable Long id) {
```

Comment:
add requestBody, PathVariable and RequestParam information in each endpoint

Reply:
not fixed, each parameter(requestBody, PathVariable and RequestParam) should have comprehensive description of waht is it for and how can it look in the request